Data download script (and associated instructions) (#180)
AlexsLemonade/refinebio-examples: CONTRIBUTING.md

Now you can navigate to http://localhost:8787/ to start developing.
Login to the RStudio server with the username `rstudio` and the password you set above.

## Download the datasets

> For developing purposes, you can download all the datasets for examples by running this command in the Docker container:

Reviewer: ```suggestion
For development purposes, you can download all the datasets for the example notebooks by running this command in the Docker container:
```

For example, if there is a `ccdl/refinebio-examples:v1` we can move to `v2`.
So the same steps would be followed above, but you can change where it says `dev` with the appropriate `v` number.

### Adding datasets to the S3 bucket

You will need an AWS account to add files to S3 bucket.
Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/) on Amazon Web Services.

Each section has a folder (`02-microarray`, `03-rnaseq`, and `04-advanced-topics`).

> 1) Click on the section folder you are adding to.

Reviewer: ```suggestion
1) Click on the folder for the section you are adding to, then on the `data` subfolder.
```

2) Click the `+ Create folder` button and name the folder with the experiment accession (e.g. `GSE12345` - as it would be on refine.bio download, aggregate by experiment).
3) Click `Save`.
4) Click on the experiment_accession folder you created.
5) Click `Upload`.
6) Choose the `metadata` and `data` TSV files you obtained from refine.bio (make sure they are correctly non-QN'ed or QN'ed upon your download).
7) Click `Next`.
8) Under `Manage public permissions` > Choose `Grant public read access to this object`.
9) Click `Next`.
10) Click `Next` again.
11) Click Upload.

Test that the files you've uploaded succesfully download by running the `download-data.sh`
script.

> If you run the script and it says `Access Denied` you may have missed step 8, but you can go back to the  file and click the `Make it Public` button and try testing your download again.

Reviewer 2: ```suggestion
If you run the script and it says `Access Denied` you may have missed step 8, but you can go back to the file and click the `Make it Public` button and try testing your download again.
```

### Adding datasets to the S3 bucket

You will need an AWS account to add files to S3 bucket.

> Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/?region=us-east-2) on Amazon Web Services.

Reviewer: ```suggestion
Go to the [refinebio-examples bucket](https://s3.console.aws.amazon.com/s3/buckets/refinebio-examples/) on Amazon Web Services.
```

- Output results should be `TSV` when possible.  
- Plots should be saved to `PNG` whenever possible.    
- Underscores instead of `-` just so we don't have to think about it.

> - No camel case.   

Author: Good point.
